src/components/ha-button-toggle-group.ts

```diff
@@ -15,22 +15,28 @@ import type { ToggleButton } from "../types";
 
 @customElement("ha-button-toggle-group")
 export class HaButtonToggleGroup extends LitElement {
-  @property() public buttons!: ToggleButton[];
+  @property({ attribute: false }) public buttons!: ToggleButton[];
 
   @property() public active?: string;
 
   protected render(): TemplateResult {
+    if (this.buttons.length <= 1) {
+      return html``;
+    }
+
     return html`
       <div>
         ${this.buttons.map(
-          (button) => html` <ha-icon-button
-            .label=${button.label}
-            .icon=${button.icon}
-            .value=${button.value}
-            ?active=${this.active === button.value}
-            @click=${this._handleClick}
-          >
-          </ha-icon-button>`
+          (button) => html`
+            <ha-icon-button
```

Comment:
We should avoid the use of `ha-icon-button` and `hass:icon` and use `ha-svg-icon` if the icon is not user configurable.